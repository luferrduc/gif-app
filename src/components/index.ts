// export { AddCategory } from "./AddCategory";
// export { GifGrid } from "./GifGrid";
// export { GifItem } from "./GifItem";
// export { Loading } from "./Loading";

export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GifItem";
export * from "./Loading";

