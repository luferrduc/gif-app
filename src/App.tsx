import { AddCategory, GifGrid } from "@/components";
import { useState } from "react"



export const App = () => {
  const [categories, setCategories] = useState<string[]>([]);

  const onAddCategory = (category: string) => {

    const isNewCategoryExists = categories?.find( cat => cat?.toLowerCase() === category.toLowerCase() )
    if(isNewCategoryExists) return
    
    setCategories([category,...categories]);
  }
  
  return (
    <>
      <h1 className="title">Gif Expert App</h1>
      {/* Form con el input para buscar */}
      
      <AddCategory onAddCategory={onAddCategory} />
      {/* Otra forma de hacerlo mandando el setState de una, no tan recomendado */}
      {/* <AddCategory setCategories={setCategories} /> */} 
      {/* Grid o listdo de gifs segÃºn el input */}

      
      {
        categories.map(category => (<GifGrid key={category} category={category}/>))
      }
 
    </>
  )
}