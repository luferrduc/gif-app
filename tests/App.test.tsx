import React from 'react'
import { fireEvent, render, screen } from "@testing-library/react"
import { App } from '../src/App'


describe('Pruebas en <App />', () => { 
  

  test('debe hacer match con el snapshot', () => { 
      
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  })


  test('debe de aparecer el texto Gif Expert App en una etiqueta h1', () => { 
    
    render(<App />);
    expect(screen.getByRole('heading', {
      level: 1
    }).innerHTML).toBe('Gif Expert App');
    

  })

  test('debe de aparecer el input vacÃ­o', () => { 
    
    render(<App />);
    const input = screen.getByRole('textbox') as HTMLInputElement;
    expect(input.value).toBe('');
    

  })


  test('debe de aparecer un h2 con el texto "One Punch" al ingresar el valor en el input y mandar el formulario', () => { 
    
    render(<App />);
    const input = screen.getByRole('textbox') as HTMLInputElement;
    const form = screen.getByRole('form') as HTMLFormElement;

    fireEvent.input(input, {
      target: {
        value: "One Punch"
      }
    })

    fireEvent.submit(form);

    expect(screen.getByRole('heading', { level: 3 }).innerHTML).toBe('One Punch');

  })
})