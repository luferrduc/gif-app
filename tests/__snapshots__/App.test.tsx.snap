// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> debe hacer match con el snapshot 1`] = `
<div>
  <h1
    class="title"
  >
    Gif Expert App
  </h1>
  <form
    aria-label="form"
    class="form"
  >
    <input
      id="input-category"
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
